pandas/io/formats/style.py

```diff
@@ -1891,11 +1891,12 @@ def set_sticky(
         else:
             # handle the MultiIndex case
             range_idx = list(range(obj.nlevels))
-            levels = sorted(levels) if levels else range_idx
+            levels_: list[int] = refactor_levels(levels, obj) if levels else range_idx
```

Comment:
wouldn't object to your moving rfactor_levels to indexing (of course could be a different PR too)

Reply:
but later :->